src/elementary/ElementID.hpp

```diff
@@ -96,4 +105,18 @@ namespace elementary {
 } // elementary namespace
 } // njoy namespace
 
+namespace std {
+
+  // std::hash override for the ElementID class
+  template <>
+  struct hash< njoy::elementary::ElementID > {
+
+    size_t operator()( const njoy::elementary::ElementID& key ) const {

```

Comment:
Why does `hash` return a `size_t` instead of an `int`?  Isn't `size_t` just an `unsigned int`?  If so, I think the `unsigned int` makes a lot more sense.  But looking at cppreference, I guess what you have is standard...

Reply:
It's indeed the standard so I copied the return type ...